signals/src/signal.rs

```diff
@@ -7,14 +7,32 @@ pub use map::*;
 pub use mutable::*;
 pub use read::*;
 
+pub type Listener = std::sync::Arc<dyn Fn(()) + Send + Sync + 'static>;
+
+/// Wraps any broadcast::ListenerGuard<T> in a type-erased container
+pub struct ListenerGuard(Box<dyn crate::broadcast::TListenerGuard + Send + Sync + 'static>);
```

Comment:
Previously, we were using broadcast::ListenerGuard<()> for all signals, which limited which broadcasts we could drive signals with to only Broadcast<()>

This didn't work for ReactorSubscription, because it uses a Broadcast<ReactorUpdate> - and we need to drive the LiveQuery Signal impl off of the ReactorSubscription in order to get content updates